### PR TITLE
Updated linter rules in cocoon

### PR DESCRIPTION
**Pull request by ricardoamador**
## Description

Updated the linter rules for auto-submit so that we use final locals and final fields.

## Pre-launch Checklist

- [x] I read the [Contributor Guide] and followed the process outlined there for submitting PRs.
- [x] I read the [Tree Hygiene] wiki page, which explains my responsibilities.
- [x] I read the [Flutter Style Guide] _recently_, and have followed its advice.
- [x] I signed the [CLA].
- [ ] I listed at least one issue that this PR fixes in the description above.
- [ ] I updated/added relevant documentation (doc comments with `///`).
- [ ] I added new tests to check the change I am making, or this PR is [test-exempt].
- [x] All existing and new tests are passing.

If you need help, consider asking for advice on the #hackers-new channel on [Discord].

[Contributor Guide]: https://github.com/flutter/flutter/wiki/Tree-hygiene#overview
[Tree Hygiene]: https://github.com/flutter/flutter/wiki/Tree-hygiene
[test-exempt]: https://github.com/flutter/flutter/wiki/Tree-hygiene#tests
[Flutter Style Guide]: https://github.com/flutter/flutter/wiki/Style-guide-for-Flutter-repo
[CLA]: https://cla.developers.google.com/
[flutter/tests]: https://github.com/flutter/tests
[breaking change policy]: https://github.com/flutter/flutter/wiki/Tree-hygiene#handling-breaking-changes
[Discord]: https://github.com/flutter/flutter/wiki/Chat

### CODE REVIEW

File: auto_submit/analysis_options.yaml [hunk 1]
```diff
@@ -9,4 +9,5 @@ linter:
     constant_identifier_names: false # we have all capitalized enums in check_for_waiting_pull_requests_test.dart
     require_trailing_commas: true
     unawaited_futures: true
-    prefer_final_fields: true
\ No newline at end of file
+    prefer_final_fields: true
+    prefer_final_locals: true
```
Comment (CaseyHillers) [inline auto_submit/analysis_options.yaml:13]: Is this something we can add in https://cs.opensource.google/flutter/cocoon/+/main:analysis_options.yaml?

Generally, we want all the projects in Cocoon to follow the same style guide

File: auto_submit/analysis_options.yaml [hunk 2]
```diff
@@ -9,4 +9,5 @@ linter:
     constant_identifier_names: false # we have all capitalized enums in check_for_waiting_pull_requests_test.dart
     require_trailing_commas: true
     unawaited_futures: true
-    prefer_final_fields: true
\ No newline at end of file
+    prefer_final_fields: true
```
Comment (CaseyHillers) [inline auto_submit/analysis_options.yaml:12]: This can be removed as it's in the parent analysis_options.yaml

**Review by CaseyHillers — APPROVED** [review]

File: auto_submit/analysis_options.yaml (hunk 1 above)
Comment (ricardoamador) [reply auto_submit/analysis_options.yaml:13]: Yeah I can add this there. No problem.

File: auto_submit/analysis_options.yaml (hunk 2 above)
Comment (ricardoamador) [reply auto_submit/analysis_options.yaml]: Done.

**Comment by mnguyen427** [comment]
Hi Ricardo! I'm looking to work on [this issue](https://github.com/flutter/flutter/issues/100074). I saw that you updated some iOS checks in this PR and was wondering how you knew what to pass into the eval function. For example, in the batteryLevelCheck method in /cocoon/device_doctor/lib/src/ios_device.dart, eval is called as follows:
`final String batteryCheckResult = await eval('ideviceinfo',
        <String>['-q', 'com.apple.mobile.battery', '-k', 'BatteryCurrentCapacity'],
        processManager: processManager,
      );`
      
I'm looking to see if it is possible to detect if screen rotation is enabled or not on iOS using eval. Thanks!

**Comment by ricardoamador** [comment]
Hello @mnguyen427 I didn't update the check just the formatting with the linter rules. @vashworth will probably know how to do this for sure. I am sure there is a way to do it similar to how android uses adb. I'm just not as familiar with the ios tools as I am with the android tools.

**Comment by vashworth** [comment]
I'm not aware of any specific tool that checks if the device has orientation locked. You can play with `ideviceinfo` to see if it has that capability. You can setup `ideviceinfo` locally (https://github.com/libimobiledevice/libimobiledevice) and run `ideviceinfo --help` to get a list of available options.

**Pull request merged**